os/arch/arm/src/amebasmart/amebasmart_pmhelpers.c

```diff
@@ -115,19 +115,29 @@ int SOCPS_AONWakeReason(void)
 
 void pg_timer_int_handler(void *Data)
 {
-	pmvdbg("Timer wakeup interrupt handler!!\n");
+	pmvdbg("PM Timer interrupt handler!!\n");
+	switch (g_pm_timer.timer_type) {
+		case PM_LOCK_TIMER:
+			pm_relax(PM_IDLE_DOMAIN, PM_NORMAL);
+			break;
+		case PM_WAKEUP_TIMER:
+			pm_activity(PM_IDLE_DOMAIN, 9);
+			break;
+		default:
+			pmdbg("Timer callback triggered without setting timer, Unexpected Error!!!\n");
+			break;
+	}
 	// Reset the global struct
-    g_timer_wakeup.use_timer = 0;
-    g_timer_wakeup.timer_interval = 0;
+	g_pm_timer.timer_type = 0;
+	g_pm_timer.timer_interval = 0;
 	// Switch status back to normal mode after wake up from interrupt
-	pm_activity(PM_IDLE_DOMAIN, 9);
 }
 
-void ap_timer_helper(void) {
+void up_set_timer(void) {
 	// Check whether timer interrupt need to be set
-	if (g_timer_wakeup.use_timer) {
+	if (g_pm_timer.timer_type) {
```

Comment:
Even pm calls the `up_set_timer`, if `g_pm_timer.timer_type` is null, then it does nothing without any warning or error. Is it ok?

Reply:
This is intended functionality, if `g_pm_timer.timer_type` is `0`, that means we do not need to invoke a timed callback along with `pm_stay()`

Reply:
Instead of timer stop, don't call the callback?

Reply:
Sometimes, the locking of PM state transition may not be time bound
i.e. we may want to lock the state transition indefinitely.
So in such cases, we don't want a timer for anything, so no timer is set at all.